woocommerce/woocommerce-android
# [Login] extract the account mismatch view to a separate screen

Part of: #7307 


### Description
This is the first step in implementing the account mismatch improvements, the extraction here will allow us two points:
1. It will make adding functionality to the account mismatch screen easier.
2. It will allow reusing the screen to fix the bug #7299.

But this turned out to be bigger than I expected 🤷.

### Testing instructions
1. Open the app and use "Enter site address" for login.
2. Enter the address of a Jetpack site that's connected to a different account.
4. Continue signing.
5. Notice the account mismatch error: if your account has other sites, it will show "View connected stores", otherwise it will show "Try another address" 

### Images/gif
|  |  |
| - | - |
| <img src="https://user-images.githubusercontent.com/1657201/187700713-3a371d2a-e290-428b-afcf-93239caf1eee.png"/> | <img src="https://user-images.githubusercontent.com/1657201/187703169-95aba5a8-f4d5-4c6a-b6a8-83bf6bbec147.png"/> |
